src/modbus-client-core.js

```diff
@@ -48,7 +48,6 @@ module.exports = stampit()
 
       currentRequest.timeout = setTimeout(function () {
         currentRequest.defer.reject({ err: 'timeout' })
-        this.emit('trashCurrentRequest')
```

Comment:
I think the currentRequestId needs to be reset here, otherwise a request can be rejected and resolved.